netmaster/master/netmaster.go

```diff
@@ -501,17 +501,26 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
 	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+	if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
> Doing so would also mean that if there is only one interface this is not needed because that is the default interface.

I think this more of a state modelling decision. I am fine tagging an endpoint with a flag or having a tenant level flag. I feel latter needs less error checking but am completely ok with either as long as we test :)

>Also note that this is only a problem when not integrating with dockcontivnet, because libnetwork takes care of adding the only interface as a default interface.

I think libnetwork also depends on the driver to return the default gateway, so we need to solve this there as well.

> This change is for bridged mode, for example in ACI,  ... We need to cover those perhaps as part of those end-to-end system tests.

I am completely ok if we are deciding to add them later. But usually it might be a good practice to back every functionality with tests otherwise we risk having breakages as more changes come in.